pyanaconda/ui/gui/spokes/network.py

```diff
@@ -1011,53 +819,58 @@ def _refresh_ap(self, dev_cfg, state=None):
             self._updating_device = False
 
     def _refresh_slaves(self, dev_cfg):
-        if dev_cfg.get_device_type() in [NM.DeviceType.BOND,
-                                         NM.DeviceType.TEAM,
-                                         NM.DeviceType.BRIDGE]:
+        if dev_cfg.device_type in [NM.DeviceType.BOND,
+                                      NM.DeviceType.TEAM,
+                                      NM.DeviceType.BRIDGE]:
             slaves = ""
-            if dev_cfg.device:
-                slaves = ",".join(s.get_iface() for s in dev_cfg.device.get_slaves())
+            device = self.client.get_device_by_iface(dev_cfg.device_name)
+            if device:
+                slaves = ",".join(s.get_iface() for s in device.get_slaves())
             self._set_device_info_value("wired", "slaves", slaves)
 
     def _refresh_parent_vlanid(self, dev_cfg):
-        if dev_cfg.get_device_type() == NM.DeviceType.VLAN:
+        if dev_cfg.device_type == NM.DeviceType.VLAN:
             parent = ""
             vlanid = ""
-            if dev_cfg.device:
-                vlanid = dev_cfg.device.get_vlan_id()
-                parent = dev_cfg.device.get_parent().get_iface()
-            elif dev_cfg.con:
-                vlanid = dev_cfg.con.get_setting_vlan().get_id()
-                parent = dev_cfg.con.get_setting_vlan().get_parent()
+            device = self.client.get_device_by_iface(dev_cfg.device_name)
+            if device:
+                vlanid = device.get_vlan_id()
+                parent = device.get_parent().get_iface()
+            else:
+                con = self.client.get_connection_by_uuid(dev_cfg.connection_uuid)
+                if con:
+                    vlanid = con.get_setting_vlan().get_id()
+                    parent = con.get_setting_vlan().get_parent()
             self._set_device_info_value("wired", "vlanid", str(vlanid))
             self._set_device_info_value("wired", "parent", parent)
 
     def _refresh_speed_hwaddr(self, dev_cfg, state=None):
-        dev_type = dev_cfg.get_device_type()
+        dev_type = dev_cfg.device_type
         if dev_type in self.wired_ui_device_types:
             dt = "wired"
         elif dev_type == NM.DeviceType.WIFI:
             dt = "wireless"
 
         # Speed
         speed = None
-        if dev_cfg.device:
+        device = self.client.get_device_by_iface(dev_cfg.device_name)
+        if device:
             if dev_type == NM.DeviceType.ETHERNET:
-                speed = dev_cfg.device.get_speed()
+                speed = device.get_speed()
             elif dev_type == NM.DeviceType.WIFI:
-                speed = dev_cfg.device.get_bitrate() / 1000
+                speed = device.get_bitrate() / 1000
```

Comment:
Could you please abstract this get speed elsewhere. For me it is easy to miss that first two if's are for how to get speed and the last one is about state.

Reply:
Hope it is more clear now.

Reply:
Looks much better to me now. Thanks!